### PR TITLE
Add more graph tests

### PR DESCRIPTION
**Pull request by marthacryan**
* Refactors some of the tests to use utils for clarity
* Adds tests for:
  * Update x axis columns
  * Update y axis columns
  * deleting the source data deletes the graph
* Fixes a bug with deleting source data

### CODE REVIEW

File: tests/streamlit_ui_tests/graph.spec.ts [hunk 1]
```diff
@@ -44,13 +44,41 @@ const testDeleteTitleThroughContextMenu = async (page, selector) => {
...
+  await mito.locator('.mito-dropdown-item', { hasText: columnName }).click();
+  await awaitResponse(page);
+}
+
+const openGraphEditor = async (mito: FrameLocator, page: Page) => {
+  await clickButtonAndAwaitResponse(page, mito, { name: 'Graph' })
+  await expect(mito.locator('.g-gtitle')).toBeVisible();
+}
+
+const changeChartType = async (mito: FrameLocator, page: Page, chartType: string, chartSubType?: string, nth?: number) => {
```
Comment (aarondr77) [inline tests/streamlit_ui_tests/graph.spec.ts:58]: It was confusing to me reading this utility when I would want to use `nth` / what it is. 

I think we use it if you are looking at a grouped bar chart to select between the `horizontal` and `vertical` ones. 

Maybe instead we could set the argument to `orientation: 'horizontal' | 'vertical' and then inside of the function we can convert those strings to 0 or 1 to be used in the nth function?

That would be easier for us to consume going forward probably.

File: tests/streamlit_ui_tests/graph.spec.ts [hunk 2]
```diff
@@ -44,13 +44,41 @@ const testDeleteTitleThroughContextMenu = async (page, selector) => {
...
 
+const addColumnToAxis = async (mito: FrameLocator, page: Page, axis: 'X' | 'Y', columnName: string) => {
+  await mito.locator('.spacing-row', { hasText: `${axis} axis` }).locator('.mito-dropdown-button').click();
+  await mito.locator('.mito-dropdown-item', { hasText: columnName }).click();
+  await awaitResponse(page);
+}
+
+const openGraphEditor = async (mito: FrameLocator, page: Page) => {
+  await clickButtonAndAwaitResponse(page, mito, { name: 'Graph' })
+  await expect(mito.locator('.g-gtitle')).toBeVisible();
```
Comment (aarondr77) [inline tests/streamlit_ui_tests/graph.spec.ts:55]: This would only pass if the before calling `openGraphEditor` you have an entire column selected, right? Because otherwise we won't generate a graph and the g-gittle won't be visible. 

Maybe we should just remove the check and add an `await awaitResponse(page);`

File: tests/streamlit_ui_tests/graph.spec.ts [hunk 3]
```diff
@@ -44,13 +44,41 @@ const testDeleteTitleThroughContextMenu = async (page, selector) => {
...
+    await mito.locator('.mito-dropdown-item-vertical').getByText(chartSubType, { exact: true }).nth(nth ?? 0).click();
+  } else {
+    await mito.getByText(chartType).click();
+  }
+  await awaitResponse(page);
+}
+
+const addChartElement = async (mito: FrameLocator, page: Page, chartElement: string, chartSubElement?: string) => {
+  await mito.getByRole('button', { name: '▾ Add Chart Element' }).click();
+  await mito.getByRole('button', { name: chartElement }).hover();
```
Comment (aarondr77) [inline tests/streamlit_ui_tests/graph.spec.ts:71]: ooo nice. Hover then click!

File: tests/streamlit_ui_tests/graph.spec.ts [hunk 4]
```diff
@@ -83,39 +111,27 @@ test.describe('Graph Functionality', () => {
...
 
-    await clickButtonAndAwaitResponse(page, mito, { name: 'Graph' })
-    await expect(mito.getByText('Column1 bar chart')).toBeVisible();
-
-    await mito.getByRole('button', { name: '▾ Change Chart Type' }).click();
-    await mito.getByText('Line').hover();
-    await clickButtonAndAwaitResponse(page, mito, { name: 'Linear' })
+    await openGraphEditor(mito, page);
 
+    await changeChartType(mito, page, 'Line', 'Linear');
```
Comment (aarondr77) [inline tests/streamlit_ui_tests/graph.spec.ts:116]: wow so nice to have this utility

File: tests/streamlit_ui_tests/graph.spec.ts [hunk 5]
```diff
@@ -167,41 +173,56 @@ test.describe('Graph Functionality', () => {
...
+    // Change the aggregation function
     await mito.locator('.mito-graph-configuration-container').getByText('count').click();
     await mito.locator('.mito-dropdown-item', { hasText: 'sum' }).click();
     await expect(mito.locator('.g-ytitle')).toHaveText('sum of None');
+
+    // Change the bin size
+    await expect(mito.locator('g.point')).toHaveCount(6);
+    await mito.locator('.mito-graph-configuration-container').locator('input[type="number"]').fill('1');
+    await awaitResponse(page);
+    await expect(mito.locator('g.point')).toHaveCount(4);
```
Comment (aarondr77) [inline tests/streamlit_ui_tests/graph.spec.ts:191]: I think some comments here to describe how we know that there should be a count of 6 and 4 would be useful

File: tests/streamlit_ui_tests/graph.spec.ts [hunk 6]
```diff
@@ -514,12 +485,97 @@ test.describe('Graph Functionality', () => {
...
+    await openGraphEditor(mito, page);
+
+    await mito.locator('.footer').getByText('test' ).click();
+    await mito.locator('.tab', { hasText: 'test'}).click({ button: 'right' });
+    await mito.locator('.mito-dropdown-item', { hasText: 'Delete' }).click();
+    await awaitResponse(page);
+    await expect(mito.getByText('Delete Sheet and Dependant Graphs')).toBeVisible();
+    await mito.getByText('Delete Sheet and Graphs').click();
+    await awaitResponse(page);
+    await expect(mito.getByRole('button', { name: 'Import Files' })).toBeVisible();
```
Comment (aarondr77) [inline tests/streamlit_ui_tests/graph.spec.ts:530]: Maybe we should check that there is not a graph tab in the footer? Here, we are only really checking that we have an empty tab

File: tests/streamlit_ui_tests/graph.spec.ts [hunk 7]
```diff
@@ -514,12 +485,97 @@ test.describe('Graph Functionality', () => {
...
+  //   await awaitResponse(page);
+  //   await expect(mito.locator('.plotly-graph-div')).toHaveCSS('width', '712px');
+
+  //   // Change the window size and check that the graph has been updated
+  //   await page.setViewportSize({ width: 1000, height: 800 });
+  //   await awaitResponse(page);
+  //   await expect(mito.locator('.plotly-graph-div')).toHaveCSS('width', '512px');
+  // });
+
+  test('Delete a column that is being used in a graph', async ({ page }) => {
```
Comment (aarondr77) [inline tests/streamlit_ui_tests/graph.spec.ts:549]: Great test!

**Review by aarondr77 — COMMENTED** [review]
Okay nice tests here! You're testing a lot of great edge case behavior that its going to be nice to not have to test manually in future releases! woohoo I'm excited about that.

File: tests/streamlit_ui_tests/graph.spec.ts (hunk 1 above)
Comment (marthacryan) [reply tests/streamlit_ui_tests/graph.spec.ts:58]: Ah yeah that's true, that's the only case that they would have the same `chartSubType` value.

File: tests/streamlit_ui_tests/graph.spec.ts (hunk 2 above)
Comment (marthacryan) [reply tests/streamlit_ui_tests/graph.spec.ts:55]: I tried that but the way that we open the graph displays stuff and doesn't include a visual loading indicator, so it registers as finished before it actually loads the graph. This causes inconsistent behavior in the tests. You're right that a column needs to be selected but I think given that the tests are generally doing that it's ok. I could put it in a wrapper with opening a specific csv so that it's guaranteed that a column is selected?

Comment (aarondr77) [reply tests/streamlit_ui_tests/graph.spec.ts:55]: Okay makes sense. Want to just add this as a comment in the utility so if we ever run into an issue with it we have this context to help us debug?
